File: factory_pattern/python/pizza_store/pizza_store.py
from abc import ABC, abstractmethod

from pizza import Pizza


class PizzaStore(ABC):

    @abstractmethod
    def create_pizza(self, pizza_type: str) -> Pizza:
        pass

    def order_pizza(self, pizza_type: str) -> Pizza:
        pizza: Pizza = self.order_pizza(pizza_type)

        pizza.prepare()
        pizza.bake()
        pizza.cut()
        pizza.box()

        return pizza

File: factory_pattern/java/pizza_store/PizzaStore.java
package pizza_store;


public abstract class PizzaStore {

    abstract Pizza createPizza(String type);

    public Pizza orderPizza(String type) {
        Pizza pizza;
        pizza = factory.createPizza(type);

        pizza.prepare();
        pizza.bake();
        pizza.cut();
        pizza.box();

        return pizza;
    }

}